sentiment_analysis.py

```
@@ -557,33 +548,23 @@ def create_interactive_dashboard(app_data, comparison_data=None):
-                            y=heatmap_pivot.index,
-                            colorscale="Viridis"
-                        ), row=5, col=1)
-
-            # Emotion-Topic Bar Charts
+                        fig.add_trace(go.Heatmap(z=heatmap_pivot.values, x=heatmap_pivot.columns, y=heatmap_pivot.index, colorscale="Viridis"), row=5, col=1)
+
             def get_top_emotion_topics(df, emotion, n=10):
                 emotion_df = df[(df["emotion"] == emotion) & (df['topics'].apply(len) > 0)]
-                if emotion_df.empty:
-                    return pd.DataFrame(columns=["Topic", "Count"])
+                if emotion_df.empty: return pd.DataFrame(columns=["Topic", "Count"])
```

Ruff (0.14.0)

Multiple statements on one line (colon)

(E701)